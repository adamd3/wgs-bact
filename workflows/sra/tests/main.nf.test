nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_fastq_ftp"
    tag "sra_ids_to_runinfo"
    tag "sra_merge_samplesheet"
    tag "sra_runinfo_to_ftp"
    tag "sra_to_samplesheet"
    tag "sra_default_parameters"

    test("Parameters: default") {

        when {
            workflow {
                """
                input[0] = Channel.from('ERR1160846', 'GSE214215', 'SRR12848126')
                """
            }
            params {
                outdir = "results"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
