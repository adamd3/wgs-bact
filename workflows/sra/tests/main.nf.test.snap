{
    "Parameters: default": {
        "content": [
            [
                "id_mappings.csv:md5,71ebb4334e4a319a1a007370e36c98f7"
            ],
            [
                "samplesheet.csv:md5,0f35f9485a633d9c581903abd611b646"
            ]
        ],
        "timestamp": "2023-08-25T14:23:33+0000"
    }
}